src/test/java/uk/gov/ons/census/casesvc/messaging/QuestionnaireLinkedReceiverIT.java

```diff
@@ -346,6 +346,110 @@ public void testGoodIndividualQuestionnaireLinked() throws Exception {
     validateEvents(events, expectedQuestionnaireId);
   }
 
+  @Test
+  public void testGoodIndividualQuestionnaireLinkedToCE() throws Exception {
+    // GIVEN
+    BlockingQueue<String> outboundUacQueue = rabbitQueueHelper.listen(rhUacQueue);
+    BlockingQueue<String> outboundCaseQueue = rabbitQueueHelper.listen(rhCaseQueue);
+
+    // Create CE (parent) case
+    Case testHHCase = easyRandom.nextObject(Case.class);
+    testHHCase.setCaseId(TEST_CASE_ID);
+    testHHCase.setCaseType("CE");
+    testHHCase.setReceiptReceived(false);
+    testHHCase.setUacQidLinks(null);
+    testHHCase.setEvents(null);
+    caseRepository.saveAndFlush(testHHCase);
+
+    // Send unaddressed uac message to create uac/qid unaddressed pair
+    CreateUacQid createUacQid = new CreateUacQid();
+    createUacQid.setQuestionnaireType("21");
+    createUacQid.setBatchId(UUID.randomUUID());
+    sendMessageAndExpectInboundMessage(unaddressedQueue, createUacQid, outboundUacQueue);
+
+    // Get generated Questionnaire Id
+    String expectedQuestionnaireId = uacQidLinkRepository.findAll().get(0).getQid();
+
+    ResponseManagementEvent managementEvent = getTestResponseManagementQuestionnaireLinkedEvent();
+    managementEvent.getEvent().setTransactionId(UUID.randomUUID());
+    UacDTO uac = new UacDTO();
+    uac.setCaseId(TEST_CASE_ID.toString());
+    uac.setQuestionnaireId(expectedQuestionnaireId);
+    managementEvent.getPayload().setUac(uac);
+
+    // WHEN
+    // Send questionnaire linked message
+    sendMessageAndDoNotExpectInboundMessage(
+        questionnaireLinkedQueue, managementEvent, outboundCaseQueue);
+
+    // THEN
+    // Check database HI Case created as expected
```

Comment:
We shouldn't be creating a `HI` case here. It looks like the code is actually checking the parent case ID so only the comment is incorrect?